src/figuresClasses.ts

```diff
@@ -1,11 +1,102 @@
-export interface Figure {}
+export enum Shape {
+  Triangle = 'triangle',
+  Rectangle = 'rectangle',
+  Circle = 'circle',
+}
+
+export enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public readonly shape: Shape = Shape.Triangle;
 
-export class Triangle implements Figure {}
+  constructor(
+    public readonly color: Color,
+    public readonly a: number,
+    public readonly b: number,
+    public readonly c: number,
+  ) {
+    if ([a, b, c].some((side) => side <= 0)) {
+      throw new Error('All sides must be > 0');
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in the error message. Instead of 'All sides must be > 0', it would be more informative to include which parameter is invalid and what the received value was.